src/core/lv2/Lv2Proc.cpp

```diff
@@ -399,6 +409,13 @@ void Lv2Proc::initPlugin()
 
 	if (m_instance)
 	{
+		const LV2_Worker_Interface* iface = (const LV2_Worker_Interface*)
+			lilv_instance_get_extension_data(m_instance, LV2_WORKER__interface);
+		if(iface) {
+			bool threaded = !Engine::audioEngine()->renderOnly();
```

Comment:
This means CLI rendering and GUI rendering will behave differently, right?

Reply:
Correct. This is explained in the [lv2 Worker spec](http://lv2plug.in/ns/ext/worker):

```
This interface is designed to gracefully support single-threaded synchronous execution, which allows
the same code to work with sample accuracy for offline rendering. For example, a sampler plugin
may schedule a sample to be loaded from disk in another thread. During live execution, the host will
call the plugin's work method from another thread, and deliver the result to the audio thread when
it is finished. However, during offline export, the scheduled load would be executed immediately in
the same thread. This enables reproducible offline rendering, where any changes affect the output
immediately regardless of how long the work takes to execute.
```

Reply:
I wrote the comment incorrectly. I was asking for differences in exports from GUI. Are workers recreated when starting export from GUI?

Reply:
> How will the plugin know if we start exporting from GUI?

For examples, for LADSPA plugins, `AudioEngine::sampleRateChanged` is connected to `LadspaEffect::changeSampleRate`, and the slot re-instantiates the plugin.

Reply:
Wouldn't it be possible that the exporting runs with the same sample rate?

Reply:
IIRC, the signal is fired in that case as well.

Reply:
It took me only 3 weeks to find out that this should be solved after #6401 will be merged :facepalm: . So I think this comment can be seen as resolved.